src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,37 @@
 package core.mate.academy.service;
 
-/**
- * Your implementation of MachineService.
- */
-public class MachineServiceImpl {
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.LinkedList;
+import java.util.List;
+
+public class MachineServiceImpl implements MachineService<Machine> {
+    @Override
+    public List getAll(Class<? extends Machine> type) {
+        if (Bulldozer.class.equals(type)) {
+            return new BulldozerProducer().get();
+        }
+        if (Excavator.class.equals(type)) {
+            return new ExcavatorProducer().get();
+        }
+        if (Truck.class.equals(type)) {
+            return new TruckProducer().get();
+        } else {
+            return new LinkedList<Machine>();
+        }
+    }
```

Comment:
```suggestion
    public List<Machine> getAll(Class<? extends Machine> type) {
        if (Bulldozer.class.equals(type)) {
            return new ArrayList(new BulldozerProducer().get());
        }
        if (Excavator.class.equals(type)) {
            return new ArrayList(new ExcavatorProducer().get());
        }
        if (Truck.class.equals(type)) {
            return new ArrayList(new TruckProducer().get());
        } else {
            return new LinkedList<Machine>();
        }
    }
```
yes, we an to this that way